pallets/eth-bridge/src/lib.rs

```diff
@@ -556,6 +549,69 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        #[pallet::call_index(7)]
+        #[pallet::weight(<T as Config>::WeightInfo::add_eth_tx_hash())]
+        pub fn submit_latest_ethereum_block(
+            origin: OriginFor<T>,
+            author: Author<T>,
+            latest_seen_block: u32,
+            _signature: <T::AuthorityId as RuntimeAppPublic>::Signature,
+        ) -> DispatchResultWithPostInfo {
+            ensure_none(origin)?;
+            ensure!(Self::active_ethereum_range().is_some(), Error::<T>::VotingEnded);
+            ensure!(
+                author_has_cast_event_vote::<T>(&author.account_id) == false,
+                Error::<T>::EventVoteExists
+            );
+
+            let nominated_range =
+                events_helpers::compute_finalised_block_range_for_latest_ethereum_block(
+                    latest_seen_block,
+                );
+            if let Some(events_partition) =
+                events_helpers::discovered_eth_events_partition_factory(nominated_range, Vec::new())
+                    .first()
+            {
+                let mut votes = EthereumEvents::<T>::get(&events_partition);
+                votes.try_insert(author.account_id).map_err(|_| Error::<T>::EventVotesFull)?;
+
+                EthereumEvents::<T>::insert(&events_partition, votes);
+
+                let mut sorted_votes: Vec<(EthereumEventsPartition, usize)> =
+                    EthereumEvents::<T>::iter()
+                        .map(|(range, votes)| (range, votes.len()))
+                        .collect();
+                sorted_votes.sort_by(|(partition_a, _votes_a), (partition_b, _votes_b)| {
+                    partition_a.range().cmp(partition_b.range())
+                });
+
+                let votes_count = sorted_votes
+                    .iter()
+                    .map(|(_range, votes)| votes)
+                    .fold(0, |acc, x| acc as usize + x);
+                let mut remaining_votes_threshold = AVN::<T>::supermajority_quorum() as usize;
+
+                if votes_count >= remaining_votes_threshold as usize {
+                    let quorum = AVN::<T>::quorum() as usize;
```

Comment:
Why do we need 2 quorums?

Reply:
we need a threshold of votes to start the consensus mechanism (supermajority_quorum) and then a `quorum` collators that have seen the same range